lib/puppet/pops/types/type_calculator.rb

```diff
@@ -644,6 +644,11 @@ def infer_Symbol(o)
   end
 
   # @api private
+  def infer_Sensitive(o)
+    PSensitiveType.new(infer(o.unwrap))
```

Comment:
this gives generalized inference, in general `infer_set` should be used to get the richest information.